src/BreakpointViewer.cpp

```diff
@@ -357,6 +354,14 @@ std::optional<AddressRange> BreakpointViewer::parseLocationField(
 	return breakpoints->getBreakpoint(*index).range;
 }
 
+QString BreakpointViewer::locationString(uint16_t begin, std::optional<uint16_t> end, int adrLen)
+{
+	return QString("%1%2%3")
+	       .arg(hexValue(begin, adrLen))
+	       .arg(end ? ":" : "")
+	       .arg(end ? hexValue(*end, adrLen) : "");
+}
+
```

Comment:
This function doesn't use any `BreakpointViewer` stuff (no member variables, nor member variables). It should be a (static) free function in the `.cpp` file.

Also, can't the `begin` and `end` parameter be merged into a single `range` parameter? I think that will also simplify some of the callers.